core/interval/int.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package interval
+
+import (
+	"sort"
+)
+
+type IntegerInterval struct {
```

Comment:
Comments on everything in this file

Reply:
Gosh I always forget about doc strings 😅